modules/battery_manager/php/battery_manager.class.inc

```diff
@@ -0,0 +1,249 @@
+ * Allows user to add, activate, and deactivate entries in the test battery
+ *
+ * PHP Version 7
+ *
+ * @category Module
+ * @package  Battery_Manager
+ * @author   Victoria Foing <victoria.foing@mcin.ca>
+ * @author   Henri Rabalais <henri.rabalais@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+class Battery_Manager extends \NDB_Menu_Filter
+{
+    public $AjaxModule   = true;
+    public $skipTemplate = true;
+
+    /**
+     * Returns true if user has access to this page.
+     *
+     * @param \User $user The user whose access is being checked
+     *
+     * @return bool
+     */
+    function _hasAccess(\User $user) : bool
+    {
+        return $user->hasPermission('battery_manager_view');
+    }
+
+    /**
+     * This acts as an Ajax enpoint that handles all action requests from the 
+     * Battery Manager Module.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    public function handle(ServerRequestInterface $request) : ResponseInterface
+    {
+        $db = \Database::singleton();
+        $user = $request->getAttribute('user');
+
+        $action = $request->getQueryParams()['action'];
+        $data   = json_decode($request->getBody()->getContents(), true);
+        switch($action) {
+        case 'getOptions':
+            return $this->response(200, $this->getOptions());
+        case 'addTest':
+            return $this->addTest($db, $user, $data);
+        case 'activateTest':
+            return $this->toggleStatus($db, $data, 'Y');
+        case 'deactivateTest':
+            return $this->toggleStatus($db, $data, 'N');
+        default:
+            $this->setup();
+            return (new \LORIS\Http\Response())
+                ->withBody(new \LORIS\Http\StringStream($this->display()));
+        }
+    }
+
+    /**
+     * Gets the data source for battery manager tests
+     *
+     * @return \LORIS\Data\Provisioner
+     */
+    function getDataProvisioner() : \LORIS\Data\Provisioner
+    {
+        $provisioner = new TestProvisioner();
+
+        $user = \User::singleton();
+
+        if ($user->hasPermission('access_all_profiles') == false) {
+            $provisioner = $provisioner->filter(
+                new \LORIS\Data\Filters\UserSiteMatch()
+            );
+        }
+        return $provisioner;
+    }
+
+    function getOptions() {
+        $options = [
+                'instruments' => \Utility::getAllInstruments(),
+                'stages'      => $this->getStageList(),
+                'subprojects' => \Utility::getSubprojectList(null),
+                'visits'      => \Utility::getVisitList(),
+                'sites'       => \Utility::getSiteList(false),
+                'firstVisits' => $this->getYesNoList(),
+                'active'      => $this->getYesNoList(),
```

Comment:
@xlecours what do you think about moving the functionality to the API module and just making API calls?

Reply:
If that's possible, I like that idea!

Reply:
@xlecours ?

Reply:
@HenriRabalais sorry, notifications were off.
I sound like a good idea. Basically, TestController needs to become an endpoint under `src/Api/Endpoints/` see https://github.com/aces/Loris/pull/4244

Reply:
@driusan @xlecours can this be resolved for now and addressed in another PR?

Reply:
@driusan i would prefer that cuz id have to do that part myself i wouldnt mind getting some extra time without delaying a major module